codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/JsonShapeDeserVisitor.java

```diff
@@ -202,15 +212,55 @@ private boolean usesExpect(Shape shape) {
     protected void deserializeUnion(GenerationContext context, UnionShape shape) {
         TypeScriptWriter writer = context.getWriter();
         Model model = context.getModel();
+        Map<String, MemberShape> members = new TreeMap<>(shape.getAllMembers());
+
+        String protocolName = context.getProtocolName();
+
+        boolean isStreamingUnion = shape.hasTrait(StreamingTrait.class);
+        boolean isRpc = protocolName.equals("aws.json-1.1") || protocolName.equals("aws.json-1.0");
```

Comment:
The breaks the encapsulation of `HttpRpcProtocolGenerator` and `DocumentVisitor`, anything protocol specific should exist in the protocol generators. We can discuss offline how we could solve this.

Reply:
Yes, let's discuss